include/openthread/platform/crypto.h

```diff
@@ -0,0 +1,425 @@
+/*
+ *  Copyright (c) 2021, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ * @brief
+ *   This file includes the platform abstraction for PSA Crypto operations.
+ */
+
+#ifndef OPENTHREAD_PLATFORM_PSA_H_
+#define OPENTHREAD_PLATFORM_PSA_H_
+
+#include <stdint.h>
+
+#include <openthread/error.h>
+#include <openthread/platform/radio.h>
+#include "psa/crypto.h"
+#include <psa/crypto_types.h>
+#include "psa/crypto_values.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef union otCryptoKey 
+{
+    const uint8_t *mKey;        ///< Pointer to the buffer containing key. NULL indicates to use `mKeyRef`.
+    uint16_t       mKeyLength;  ///< The key length in bytes (applicable when `mKey` is not NULL).
+    otMacKeyRef    mKeyRef;     ///< The PSA key ref (requires `OPENTHREAD_CONFIG_PSA_CRYPTO_ENABLE` and `mKey` to be NULL).
```

Comment:
How about we just use uint32 instead? otMacKeyRef is just a typedef for uint32 after all. I feel it will be a good idea to group all the otMacKey.. datatypes together.